src/callbacks/onPreprocess/flattenData/getMaxValues.js

```diff
@@ -0,0 +1,100 @@
+import addParticipantLevelMetadata from './addParticipantLevelMetadata';
+import calculateRRatio from './calculateRRatio';
+
+export default function getMaxValues(d) {
+    var chart = this;
+    var config = this.config;
+
+    var participant_obj = {};
+    participant_obj.days_x = null;
+    participant_obj.days_y = null;
+    Object.keys(config.measure_values).forEach(function(mKey) {
+        //get all raw data for the current measure
+        var matches = d
+            .filter(f => config.measure_values[mKey] == f[config.measure_col]) //get matching measures
+            .filter(f => f.analysisFlag);
+
+        if (matches.length == 0) {
+            if (config.debug) {
+                console.warn(
+                    'No analysis records found for ' + d[0][config.id_col] + ' for ' + mKey
+                );
+            }
+
+            participant_obj.drop_participant = true;
+            participant_obj.drop_reason =
+                'No analysis results found for 1+ key measure, including ' + mKey + '.';
+            return participant_obj;
+        } else {
+            participant_obj.drop_participant = false;
+        }
+
+        //keep an array of all [value, studyday] pairs for the measure
+        participant_obj[mKey + '_raw'] = matches.map(function(m) {
+            return { value: m[config.display], day: m[config.studyday_col] };
+        });
+
+        //get the current record for each participant
+        if (config.plot_max_values) {
+            //get record with maximum value for the current display type
+            participant_obj[mKey] = d3.max(matches, d => +d[config.display]);
+            console.log('got max val');
+        } else {
+            //get the most recent measure before config.plot_day
+            var getLastMeasureIndex = d3.bisector(d => d.day).left;
+            var lastMeasureIndexPlusOne = getLastMeasureIndex(
+                participant_obj[mKey + '_raw'],
+                config.plot_day
+            );
+            var lastMeasureIndex = lastMeasureIndexPlusOne - 1;
```

Comment:
This method seems to kick out the first record if the first record is the only record less than or equal to `config.plot_day`.